EKF/control.cpp

```diff
@@ -383,8 +383,8 @@ void Ekf::controlOpticalFlowFusion()
 		_imu_del_ang_of += _imu_sample_delayed.delta_ang - _state.gyro_bias;
 		_delta_time_of += _imu_sample_delayed.delta_ang_dt;
 
-		// fuse the data
-		if (_control_status.flags.opt_flow) {
+		// fuse the data if the terrain/distance to bottom is valid
+		if (_control_status.flags.opt_flow && get_terrain_valid()) {
```

Comment:
This is going to cause problems on ground when the range finder is out of range

Reply:
We could do something like
```
// use rng_gnd_clearance if on ground
				if (!_ekf.get_in_air_status()) {
					range_finder.current_distance = _params->rng_gnd_clearance;

				// check if distance sensor is within working boundarie
				} else if (range_finder.min_distance >= range_finder.current_distance ||
					         range_finder.max_distance <= range_finder.current_distance) {
					range_finder_updated = false;
				}
```
in the Firmware. But then handheld tests wouldn't work.

Reply:
I was thinking that if on ground, an absence of range data for longer than TBD or range data below min or above some reasonable max value for hand carriage would cause rng_gnd_clearance to be used.

Reply:
or actually this way you can still do handheld tests
```
// check if distance sensor is within working boundaries
if (range_finder.min_distance >= range_finder.current_distance ||
	range_finder.max_distance <= range_finder.current_distance) {
	// use rng_gnd_clearance if on ground
	if (_ekf.get_in_air_status()) {
		range_finder_updated = false;

	} else {
		range_finder.current_distance = _params->rng_gnd_clearance;
	}
}
```